go/cmd/dolt/commands/rebase.go

```diff
@@ -0,0 +1,369 @@
+	}
+	if apr.Contains(cli.ContinueFlag) {
+		args = append(args, "'--continue'")
+	}
+	if apr.Contains(cli.AbortParam) {
+		args = append(args, "'--abort'")
+	}
+
+	/*if apr.ContainsAny(cli.ContinueFlag, cli.AbortParam) {
+		checkoutQuery = fmt.Sprintf("CALL DOLT_CHECKOUT('dolt_rebase_%s'); ", rebaseBranchName)
+	}*/
+
+	query := fmt.Sprintf("%sCALL DOLT_REBASE(%s);", checkoutQuery, strings.Join(args, ", "))
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
+	if err != nil {
+		return "", err
+	}
+
+	return interpolatedQuery, nil
+}
+
+// getRebasePlan opens an editor for users to edit the rebase plan and returns the parsed rebase plan from the editor.
+func getRebasePlan(cliCtx cli.CliContext, sqlCtx *sql.Context, queryist cli.Queryist, rebaseBranch, currentBranch string) (*rebase.RebasePlan, error) {
+	if cli.ExecuteWithStdioRestored == nil {
+		return nil, nil
+	}
+
+	if !checkIsTerminal() {
+		return nil, nil
+	}
+
+	initialRebaseMsg, err := buildInitialRebaseMsg(sqlCtx, queryist, rebaseBranch, currentBranch)
+	if err != nil {
+		return nil, err
+	}
+
+	backupEd := "vim"
+	// try getting default editor on the user system
+	if ed, edSet := os.LookupEnv(dconfig.EnvEditor); edSet {
+		backupEd = ed
+	}
+	// try getting Dolt config core.editor
+	editorStr := cliCtx.Config().GetStringOrDefault(config.DoltEditor, backupEd)
+
+	var rebaseMsg string
+	cli.ExecuteWithStdioRestored(func() {
+		rebaseMsg, err = editor.OpenTempEditor(editorStr, initialRebaseMsg)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	rebasePlan, err := parseRebaseMessage(rebaseMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	return rebasePlan, nil
+}
+
+// buildInitialRebaseMsg builds the initial message to display to the user when they open the rebase plan editor,
+// including the formatted rebase plan.
+func buildInitialRebaseMsg(sqlCtx *sql.Context, queryist cli.Queryist, rebaseBranch, currentBranch string) (string, error) {
+	var buffer bytes.Buffer
+
+	rows, err := GetRowsForSql(queryist, sqlCtx, "SELECT action, commit_hash, commit_message FROM dolt_rebase ORDER BY rebase_order")
+	if err != nil {
+		return "", err
+	}
+
+	// rebase plan
+	for _, row := range rows {
+		action, found := getRebaseAction(row[0])
+		if !found {
+			return "", errors.New("invalid rebase action")
+		}
+		commitHash := row[1].(string)
+		commitMessage := row[2].(string)
+		buffer.WriteString(fmt.Sprintf("%s %s %s\n", action, commitHash, commitMessage))
+	}
+	buffer.WriteString("\n")
+
+	// help text
+	rebaseBranchHash, err := getHashOf(queryist, sqlCtx, rebaseBranch)
+	if err != nil {
+		return "", err
+	}
+	currentBranchHash, err := getHashOf(queryist, sqlCtx, currentBranch)
+	if err != nil {
+		return "", err
+	}
+	rows, err = GetRowsForSql(queryist, sqlCtx, "SELECT COUNT(*) FROM dolt_rebase")
+	if err != nil {
+		return "", err
+	}
+	numSteps, err := getInt64ColAsInt64(rows[0][0])
+	if err != nil {
+		return "", err
+	}
```

Comment:
I think you can avoid another round trip/query to the sql engine here... if you look at the len(rows) that should give you the count of the steps in the rebase plan.